packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/TypeFactory.cs

```diff
@@ -111,18 +111,19 @@ public TypeProvider CreateModel(InputModelType model)
         /// Factory method for creating a <see cref="TypeProvider"/> based on an <see cref="InputEnumType"> <paramref name="enumType"/>.
         /// </summary>
         /// <param name="enumType">The <see cref="InputEnumType"/> to convert.</param>
+        /// <param name="declaringType"/> The declaring <see cref="TypeProvider".</param>
         /// <returns>An instance of <see cref="TypeProvider"/>.</returns>
-        public TypeProvider CreateEnum(InputEnumType enumType)
+        public TypeProvider CreateEnum(InputEnumType enumType, TypeProvider? declaringType = null)
         {
             if (CSharpToTypeProvider.TryGetValue(enumType, out var enumProvider))
```

Comment:
What happens if someone passes in the same InputEnumType with a different declaringType?  Should the cache have different values for these two calls?

Reply:
That is a good question. I personally think we should cache based on both the declaring type and input enum type. I think it's a valid scenario that someone would want to construct an enum provider using the same input type, but different declaring types. I implemented a solution for this, but I'm curious to hear your thoughts on this and if there's a more performant way of doing this, or if it's even necesary..

Reply:
lets use a readonly struct as the key